mlir/lib/Target/IRDLToCpp/Templates/PerOperationDecl.txt

```diff
@@ -0,0 +1,137 @@
+R"(
+__NAMESPACE_OPEN__
+
+namespace detail {
+
+class __OP_CPP_NAME__GenericAdaptorBase {
+public:
+  struct Properties {
+  };
+protected:
+  ::mlir::DictionaryAttr odsAttrs;
+  ::std::optional<::mlir::OperationName> odsOpName;
+  Properties properties;
+  ::mlir::RegionRange odsRegions;
```

Comment:
The convention in LLVM is to put fields at the end of the classes, after the methods.